File: go/models/agent.go
package models

type AgentScanRequest struct {
	AgentConfig map[string]interface{} `json:"agent_config"`
	Tools       []AgentTool            `json:"tools,omitempty"`
	Prompt      string                 `json:"prompt,omitempty"`
}

type AgentTool struct {
	Name        string `json:"name"`
	Description string `json:"description"`
	Schema      string `json:"schema,omitempty"`
}

type AgentScanResponse struct {
	Safe           bool           `json:"safe"`
	Vulnerabilities []AgentVuln   `json:"vulnerabilities"`
	RiskScore      float64        `json:"risk_score"`
	Recommendations []string      `json:"recommendations"`
}

type AgentVuln struct {
	Type     string  `json:"type"`
	Severity string  `json:"severity"`
	Detail   string  `json:"detail"`
	Location string  `json:"location,omitempty"`
	Score    float64 `json:"score"`
}

type ToolchainRequest struct {
	Tools    []AgentTool `json:"tools"`
	Workflow string      `json:"workflow,omitempty"`
}

type ToolchainResponse struct {
	Safe           bool      `json:"safe"`
	Risks          []AgentVuln `json:"risks"`
	ChainAnalysis  string    `json:"chain_analysis"`
}

type MemoryPoisoningRequest struct {
	Memory  []MemoryEntry `json:"memory"`
	Query   string        `json:"query,omitempty"`
}

type MemoryEntry struct {
	Role    string `json:"role"`
	Content string `json:"content"`
}

type MemoryPoisoningResponse struct {
	Poisoned   bool      `json:"poisoned"`
	Confidence float64   `json:"confidence"`
	Flagged    []int     `json:"flagged"`
	Details    string    `json:"details"`
}

type ReasoningHijackRequest struct {
	Prompt    string `json:"prompt"`
	Response  string `json:"response"`
}

type ReasoningHijackResponse struct {
	Hijacked   bool    `json:"hijacked"`
	Confidence float64 `json:"confidence"`
	Type       string  `json:"type,omitempty"`
	Details    string  `json:"details"`
}

type ToolDisguiseRequest struct {
	ToolCall    string `json:"tool_call"`
	Description string `json:"description"`
}

type ToolDisguiseResponse struct {
	Disguised  bool    `json:"disguised"`
	Confidence float64 `json:"confidence"`
	ActualTool string  `json:"actual_tool,omitempty"`
	Details    string  `json:"details"`
}

File: rust/src/models/agent.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct AgentScanRequest {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub prompt: Option<String>,
    #[serde(default)]
    pub external_data: Vec<String>,
    #[serde(default)]
    pub session_id: String,
    #[serde(default)]
    pub user_id: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub agent_type: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub context: Option<serde_json::Value>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct AgentScanResponse {
    #[serde(default)]
    pub safe: bool,
    #[serde(default)]
    pub threats: Vec<ThreatDetection>,
    #[serde(default)]
    pub overall_risk: f64,
    #[serde(default)]
    pub recommendations: Vec<String>,
    #[serde(default)]
    pub latency_ms: f64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct ThreatDetection {
    #[serde(default)]
    pub threat_type: String,
    #[serde(default)]
    pub severity: String,
    #[serde(default)]
    pub confidence: f64,
    #[serde(default)]
    pub description: String,
    #[serde(default)]
    pub evidence: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct ToolchainRequest {
    #[serde(default)]
    pub tools: Vec<ToolDefinition>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub execution_plan: Option<Vec<String>>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub context: Option<serde_json::Value>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct ToolDefinition {
    pub name: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub description: Option<String>,
    #[serde(default)]
    pub permissions: Vec<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub parameters: Option<serde_json::Value>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct ToolchainResponse {
    #[serde(default)]
    pub safe: bool,
    #[serde(default)]
    pub risk_score: f64,
    #[serde(default)]
    pub violations: Vec<ToolchainViolation>,
    #[serde(default)]
    pub recommendations: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct ToolchainViolation {
    pub tool: String,
    #[serde(default)]
    pub violation_type: String,
    #[serde(default)]
    pub severity: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub description: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct MemoryPoisoningRequest {
    pub content: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub memory_context: Option<Vec<String>>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct MemoryPoisoningResponse {
    #[serde(default)]
    pub poisoning_detected: bool,
    #[serde(default)]
    pub confidence: f64,
    #[serde(default)]
    pub poisoned_entries: Vec<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub details: Option<serde_json::Value>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct ReasoningHijackRequest {
    pub content: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub reasoning_chain: Option<Vec<String>>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct ReasoningHijackResponse {
    #[serde(default)]
    pub hijack_detected: bool,
    #[serde(default)]
    pub confidence: f64,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub hijack_type: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub details: Option<serde_json::Value>,
}
